applications/tari_merge_mining_proxy/src/block_template_data.rs

```diff
@@ -37,18 +60,43 @@ impl BlockTemplateRepository {
     }
 
     pub async fn get<T: AsRef<[u8]>>(&self, hash: T) -> Option<BlockTemplateData> {
+        trace!(
+            target: LOG_TARGET,
+            "Retrieving blocktemplate with merge mining hash: {:x?}",
+            hash.as_ref()
+        );
         let b = self.blocks.read().await;
-        b.get(hash.as_ref()).cloned()
+        match b.get(hash.as_ref()).cloned() {
+            Some(item) => Some(item.data),
+            None => None,
+        }
     }
 
     pub async fn save<T: AsRef<[u8]>>(&mut self, hash: T, block_template: BlockTemplateData) {
+        trace!(
+            target: LOG_TARGET,
+            "Saving blocktemplate with merge mining hash: {:x?}",
+            hash.as_ref()
+        );
         let mut b = self.blocks.write().await;
-        b.insert(Vec::from(hash.as_ref()), block_template);
+        let repository_item = BlockTemplateRepositoryItem::new(block_template);
+        b.insert(Vec::from(hash.as_ref()), repository_item);
     }
 
-    pub async fn remove<T: AsRef<[u8]>>(&mut self, hash: T) -> Option<BlockTemplateData> {
+    pub async fn remove_outdated(&mut self) {
+        trace!(target: LOG_TARGET, "Removing outdated blocktemplates");
         let mut b = self.blocks.write().await;
-        b.remove(hash.as_ref())
+        let threshold = Utc::now() - Duration::minutes(20);
+        for item in b.clone().iter() {
+            if item.1.datetime() < threshold {
+                trace!(
+                    target: LOG_TARGET,
+                    "Blocktemplate removed with merge mining hash {:x?}",
+                    item.0
```

Comment:
Not printing as hex:
```
2021-01-13 12:15:58.230387500 [tari_mm_proxy::xmrig] TRACE Blocktemplate removed with merge mining hash 
[c5, db, 45, f7, e2, 36, c9, 6a, 8c, cd, 10, 85, 1b, 4, e8, 79, 1e, b0, 67, 52, 82, eb, 8a, 7b, a7, a3, 1a, 5c, 6b, 99, 2c, c8] 
```

Reply:
See above comment, will sort it out for all